crates/dojo-lang/src/semantics/utils.rs

```diff
@@ -0,0 +1,201 @@
+use std::collections::{HashMap, HashSet};
+
+use cairo_lang_compiler::db::RootDatabase;
+use cairo_lang_defs::db::DefsGroup;
+use cairo_lang_defs::ids::{FunctionWithBodyId, LookupItemId, ModuleId, ModuleItemId};
+use cairo_lang_filesystem::ids::FileId;
+use cairo_lang_lowering::db::LoweringGroup;
+use cairo_lang_lowering::ids::{self as low, SemanticFunctionWithBodyIdEx};
+use cairo_lang_lowering::Statement;
+use cairo_lang_semantic as semantic;
+use cairo_lang_syntax::node::{ast, SyntaxNode, TypedSyntaxNode};
+use semantic::db::SemanticGroup;
+use semantic::diagnostic::SemanticDiagnostics;
+use semantic::expr::compute::{ComputationContext, Environment};
+use semantic::expr::inference::InferenceId;
+use semantic::items::function_with_body::SemanticExprLookup;
+use semantic::resolve::Resolver;
+use semantic::FunctionId;
+
+use crate::inline_macros::utils::SystemRWOpRecord;
+
+pub fn find_module_rw(
+    db: &RootDatabase,
+    module_id: &ModuleId,
+    module_writers: Option<&HashMap<String, Vec<SystemRWOpRecord>>>,
```

Comment:
```suggestion
    module_writers: &HashMap<String, Vec<SystemRWOpRecord>>,
```

I think it would be cleaner to handle the option check at the call site